apps/rush-lib/src/start.ts

```diff
@@ -1,6 +1,25 @@
 // Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
 // See LICENSE in the project root for license information.
 
+import * as colors from 'colors';
+import * as semver from 'semver';
+
+// check to ensure that we are using a supported version of NodeJS, otherwise write a warning
+if (semver.satisfies(process.versions.node, '< 6.0.0')) {
+  console.error(colors.red(`You are using an outdated version of Node ("${process.versions.node}").`
+    + ` You should upgrade to Node 6 or Node 8.`));
+  process.exit(1);
+} else if (semver.satisfies(process.versions.node, '<= 6.4.0')) {
+  console.warn(colors.yellow(`You are using an outdated version of Node ("${process.versions.node}").`
+    + ` You should upgrade to Node >=6.5.0 or Node 8, as some Rush features may not work.`));
+} else if (semver.satisfies(process.versions.node, '^7.0.0')) {
+  console.warn(colors.yellow(`You are using a non-LTS version of Node ("${process.versions.node}").`
+    + ` You should consider upgrading to Node 8 or downgrading to Node 6, as some Rush features may not work.`));
+} else if (semver.satisfies(process.versions.node, '^9.0.0')) {
```

Comment:
This is the "known not to be LTS" scenario:

"Your version of Node.js (1.2.3) is not a Long Term Stable (LTS) version.  These releases are experimental and the Rush team will not accept bug reports for them.  Please consider installing a stable release."